### PR TITLE
fix: better defaulting of the git kind + webhook in boot

### PR DESCRIPTION
**Pull request by jstrachan**
based on folks using standard public git servers

### CODE REVIEW

File: pkg/config/install_requirements.go [hunk 1]
```diff
@@ -425,24 +425,28 @@ func (c *RequirementsConfig) IsLazyCreateSecrets(flag string) (bool, error) {
...
 		c.Cluster.GitName = c.Cluster.GitKind
 	}
 	if c.Ingress.NamespaceSubDomain == "" {
 		c.Ingress.NamespaceSubDomain = "-" + c.Cluster.Namespace + "."
 	}
+	if c.Webhook == WebhookTypeNone {
+		if c.Cluster.GitServer == "https://github.com" || c.Cluster.GitServer == "https://github.com/" {
+			c.Webhook = WebhookTypeProw
+		} else {
+			c.Webhook = WebhookTypeLighthouse
```
Comment (pmuir) [inline pkg/config/install_requirements.go]: I'm not sure this is a good idea without a strong warning to users that lighthouse is experimental. Can we add a confirmation dialog (override in batch mode) that says that we don't support their server type but that we do have an experimental feature called lighthouse that they could try out and make it really clear it's experimental!. Probably should be an error message to start with for the wrong type of git.

Comment (jstrachan) [reply pkg/config/install_requirements.go]: this function has no way to know if its got batch mode enabled so gonna just omit defaulting for lighthouse for now until its GA

**Pull request merged**